### PR TITLE
Fix FTBFS on Fedora and refresh RPM packaging

### PR DESCRIPTION
**Pull request by Conan-Kudo**
This pull request fixes Netplan to build on Fedora Linux and updates the RPM packaging to use the Meson build.

### CODE REVIEW

**Review by slyon — COMMENTED** [review]
Thank you! I need to double check the systemd usrmerge situation in Debian/Ubuntu wrt. that meson.build change. Otherwise this LGTM!

File: src/meson.build [hunk 1]
```diff
@@ -36,7 +36,7 @@ executable(
     install: true)
 install_symlink(
     'netplan',
-    pointing_to: join_paths('..', '..', '..') + join_paths(get_option('prefix'), libexec_netplan, 'generate'),
+    pointing_to: join_paths('..', '..', '..', '..') + join_paths(get_option('prefix'), libexec_netplan, 'generate'),
     install_dir: systemd_generator_dir)
```
Comment (xnox) [inline src/meson.build:40]: that's not quite right. as systemd_generator_dir can and is of different length on different platforms.

maybe install_symlink accepts absolute locations which then can be "optimized" automatically, or later on by packaging itself? (i.e. in debian debhelper fixes up absolute symlinks to relative ones) Doing absolute target is better here imho.

Comment (Conan-Kudo) [reply src/meson.build:40]: I'll adjust this to use absolute paths.

**Comment by slyon** [comment]
Ugh.. apparently meson is not willing to install a dangling symlink, so I needed to revert this https://github.com/canonical/netplan/commit/ea5530c9664c69f7c25d09cc10c63f5ed7a89990 and we'll need come up with something better.

**Comment by Conan-Kudo** [comment]
You might have to use a custom handler for installing symlinks instead of Meson's own thing. There's too many papercuts with Meson's `install_symlink()` method.

**Comment by slyon** [comment]
Yeah... I'm currently investigating a few options/workarrounds, like `custom_target` or `add_install_script`, as described in https://github.com/mesonbuild/meson/issues/1602

**Pull request merged**